### ISSUE TITLE
Passwords will be resetted to be five stars 

### ISSUE THREAD

**Issue by kikmedia** [contao/contao#2383]
**Affected version(s)**

Contao 4.10.2

**Description**

Today me and some collegues stumbled into an issue where passwords of adminstrative users where changed into "*****" (five stars, and I'm not kidding) after changing something in user settings.

For illustration, please see my message at https://twitter.com/carolinakoehn/status/1311640515234926592 (where you may find a screenshot of my database), the following thread, and the ticket at https://github.com/terminal42/contao-easy_themes/issues/47.

Long story short: It seems that the mask with five stars will be saved directly into tl_user, leaving one unable to log in without creating a new admin user via install tool.

We checked our installed extensions, and at the moment we think that this could be a core issue.

**Comment by Ainschy** [comment]
Same her in a 4.10.2

**Comment by leofeyer** [comment]
The issue is caused by the changes in 4d71a92293e1d0fbe31326a454b0b6d16f32b633 (see #2261). The `Password` widget changes the return value of the `submitInput()` method in the `validate()` class, however, we are now calling `submitInput()` **before** calling `validate()`.

**Comment by leofeyer** [comment]
This affects the following widgets in the core:

* `FormPassword`
* `Password`
* `TextStore`

They all change `$this->blnSubmitInput` in the `validator()` method.
